modules/mod-deep-learning/ModelCardPanel.cpp

```diff
@@ -0,0 +1,518 @@
+                                       Verbatim("%s").Format(wxString(mCard->short_description())),
+                                       false, wxLEFT);
+   // S.EndStatic();
+}
+
+void ModelCardPanel::PopulateLongDescription(ShuttleGui &S)
+{
+   // model description
+   S.StartStatic(Verbatim(""));
+   mLongDescription = S.AddVariableText(
+                                       Verbatim("%s").Format(wxString(mCard->long_description())),
+                                       false, wxLEFT, 
+                                       GetSize().GetWidth() - 30);
+   S.EndStatic();
+}
+
+void ModelCardPanel::PopulateMetadata(ShuttleGui &S)
+{
+   S.StartMultiColumn(2, wxALIGN_LEFT);
+   {
+      S.AddVariableText(XO("Effect: "))
+          ->SetFont(wxFont(wxFontInfo().Bold()));
+      S.AddVariableText(Verbatim("%s")
+                            .Format(mCard->effect_type()));
+
+      S.AddVariableText(XO("Sample Rate: "))
+          ->SetFont(wxFont(wxFontInfo().Bold()));
+      S.AddVariableText(Verbatim("%d")
+                            .Format(mCard->sample_rate()));
+
+      std::string tagString;
+      for (auto &tag : mCard->tags())
+      {
+         if (!tagString.empty())
+            tagString = tagString +  ", " + tag;
+         else
+            tagString = tag;
+      }
+      
+      S.AddVariableText(XO("Tags: "))
+          ->SetFont(wxFont(wxFontInfo().Bold()));
+      S.AddVariableText(Verbatim("%s")
+                            .Format(tagString));
+   }
+   S.EndMultiColumn();
+}
+
+void ModelCardPanel::PopulateInstallCtrls(ShuttleGui &S)
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+
+   S.StartVerticalLay(wxCENTER, true);
+   {
+      // add a progress gauge for downloads, but hide it
+      mInstallProgressGauge = safenew wxGauge(S.GetParent(), wxID_ANY, 100); // TODO:  sizing
+      mInstallProgressGauge->SetSize(wxSize(80, 20));
+      S.AddWindow(mInstallProgressGauge);
+
+      bool installed = manager.IsInstalled(mCard);
+      TranslatableString status = installed ? XO("installed") : XO("uninstalled");
+      mInstallStatusText = S.AddVariableText(status, true);
+
+      InstallStatus iStatus = installed ? InstallStatus::Installed : InstallStatus::Uninstalled;
+      wxColour statusColor = mInstallStatusColors[iStatus];
+      mInstallStatusText->SetForegroundColour(statusColor);
+
+      // TODO: do translatable strings here from the begginign
+      TranslatableString cmd = installed ? XO("Uninstall") : XO("Install");
+      mInstallButton = S.AddButton(cmd);
+
+      SetInstallStatus(installed ? InstallStatus::Installed : InstallStatus::Uninstalled);
+
+      mSelectButton = S.AddButton(XC("Select", "model"));
+      mSelectButton->Connect(wxEVT_COMMAND_BUTTON_CLICKED, 
+                        wxCommandEventHandler(ModelCardPanel::OnSelect), NULL, this);
+   }
+   S.EndVerticalLay();
+}
+
+void ModelCardPanel::PopulateMoreInfo(ShuttleGui &S)
+{
+   S.StartHorizontalLay(wxCENTER, true);
```

Comment:
This is not properly balanced with `EndHorizontalLay` !

Reply:
4164841bf